operator/internal/manifests/internal/alerts/build_test.go

```diff
@@ -0,0 +1,265 @@
+    "annotations":
+      "message": |-
+        {{ $labels.container }} memory usage is high: {{ $value }}%
+      "summary": "Loki's memory usage is high (above 95% of the configured limit for 5 minutes)"
+      "runbook_url": "TBD"
+    expr: |
+      sum(
+        container_memory_working_set_bytes{container=~"loki-.*"}
+      ) by (namespace, container)
+      /
+      sum(
+        kube_pod_container_resource_limits{resource="memory"}
+      ) by (namespace, container)
+      * 100
+      > 95
+    "for": 5m
+    "labels":
+      "severity": critical
+  - "alert": LokiStorageSlowWrite
+    "annotations":
+      "message": "The 99th percentile ({{ $value }}s) is experiencing slow write"
+      "summary": "The 99th percentile ({{ $value }}s) is experiencing slow write"
+      "runbook_url": "TBD"
+    expr: |
+      histogram_quantile(.99,
+          sum(
+              rate(
+                  loki_boltdb_shipper_request_duration_seconds_bucket{operation="WRITE"}[5m]
+              )
+          ) by (le, namespace, operation)
+      )
+      > 1
+    "for": 5m
+    "labels":
+      "severity": warning
+  - "alert": LokiStorageSlowRead
+    "annotations":
+      "message": "The 99th percentile ({{ $value }}s) is experiencing slow read"
+      "summary": "The 99th percentile ({{ $value }}s) is experiencing slow read"
+      "runbook_url": "TBD"
+    expr: |
+      histogram_quantile(.99,
+          sum(
+              rate(
+                  loki_boltdb_shipper_request_duration_seconds_bucket{operation="QUERY"}[5m]
+              )
+          ) by (le, namespace, operation)
+      )
+      > 1
+    "for": 5m
+    "labels":
+      "severity": warning
+  - "alert": LokiWritePathHighLoad
+    "annotations":
+      "message": |-
+        The write path of Loki is loaded: {{ printf "%0.f" $value }} B/s
+      "summary": "The write path of Loki is loaded"
+      "runbook_url": "TBD"
+    expr: |
+      sum(
+          rate(
+              loki_distributor_bytes_received_total{}[5m]
+          )
+      ) by (namespace)
+      >
+      1e+06
+    "for": 5m
+    "labels":
+      "severity": warning
+  - "alert": LokiReadPathHighLoad
+    "annotations":
+      "message": "The read path of Loki is loaded: {{ $value }} QPS"
+      "summary": "The read path of Loki is loaded"
+      "runbook_url": "TBD"
+    # HTTP & GRPC routes
+    expr: |
+      sum(
+          rate(
+              loki_request_duration_seconds_count{
+                  job="query-frontend",
+                  route=~"loki_api_v1_series|api_prom_series|api_prom_query|api_prom_label|api_prom_label_name_values|loki_api_v1_query|loki_api_v1_query_range|loki_api_v1_labels|loki_api_v1_label_name_values|/logproto.Querier/Query|/logproto.Querier/Label|/logproto.Querier/Series|/logproto.Querier/QuerySample|/logproto.Querier/GetChunkIDs",
+              }[5m]
+          )
+      ) by (namespace)
+      >
+      1
+    "for": 5m
+    "labels":
+      "severity": warning
+`
+	expRules := `
```

Comment:
Not sure about those alert the threshold seems too low specially for read 15s is often possible. On grafana side we usually also account for the byte_througput we expect at least 4gb/s

For write 1s is fine, but you should look at distributor latency no boltdb (same for reads)

I also suggest to look at per ingester stream it shouldn't be more than 80k

Reply:
I can't share the metric because they are taken from our gateway.

For the writes I would only look at the http request on the distributor level

Reply:
AFAIU, the distributors don't access the storage. So, are you suggesting to remove the slow storage alerts?